src/zimrecreate.cpp

```diff
@@ -31,6 +31,59 @@
 #include "tools.h"
 #include "version.h"
 
+/**
+ * A PatchItem. This patch html and css content to remove the namespcae from the links.
+ */
+class PatchItem : public zim::writer::Item
+{
+    //article from an existing ZIM file.
+    zim::Item item;
+
+  public:
+    explicit PatchItem(const zim::Item item):
+      item(item)
+    {}
+
+    virtual std::string getPath() const
+    {
+      auto path = item.getPath();
+      if (path.length() > 2 && path[1] == '/') {
+        path = path.substr(2, std::string::npos);
+      }
+      return path;
+    }
+
+    virtual std::string getTitle() const
+    {
+        return item.getTitle();
+    }
+
+    virtual std::string getMimeType() const
+    {
+        return item.getMimetype();
+    }
+
+    std::unique_ptr<zim::writer::ContentProvider> getContentProvider() const
+    {
+        auto mimetype = getMimeType();
+        if ( mimetype.find("text/html") == std::string::npos
+          && mimetype.find("text/css") == std::string::npos) {
+            return std::unique_ptr<zim::writer::ContentProvider>(new ItemProvider(item));
+        }
+
+        std::string content = item.getData();
+        // This is a really poor url rewriting to remove the starting "../<NS>/"
+        // and replace the "../../<NS/" by "../"
+        for (std::string prefix: {"'", "\""}) {
+          for (auto ns : {'A','I','J','-'}) {
+            replaceStringInPlace(content, prefix+"../../"+ns+"/", prefix+"../");
+            replaceStringInPlace(content, prefix+"../"+ns+"/", prefix);
+          }
+        }
```

Comment:
What does *poor* here imply? If you mean that it can inadvertently modify content that looks like mark-up URL, you better make that explicit.

Reply:
I've add a better comment. (And directly squash/rebase as it is the only change)